### PR TITLE
Upgrade toolchain

### PR DESCRIPTION
**Pull request by fjtrujy**
## Description
This PR basically does 2 things, improve Buildsystem scripts and upgrade toolchain.

1. Upgrade toolchain:
- Upgrade GCC to 13.2.0
- Upgrade newlib to 4.4.0

2. Buildsystem improvements (credits to @uyjulian)
- Some cleanups and normalization.
- The major change is repository URLs and references are now in a separate script to make it easier to update.

### CODE REVIEW

File: scripts/003-newlib.sh [hunk 1]
```diff
@@ -1,16 +1,37 @@
 #!/bin/bash
-# newlib.sh by Francisco Javier Trujillo Mata (fjtrujy@gmail.com)
+# 003-newlib.sh by pspdev developers
+
+## Exit with code 1 when any command executed returns a non-zero exit code.
+onerr()
```
Comment (sharkwouter) [inline scripts/003-newlib.sh:5]: Wouldn't `set -e` do the same thing?

Comment (uyjulian) [reply scripts/003-newlib.sh:5]: This is done for future extensibility

**Comment by wally4000** [comment]
Daedalus builds and works with this change.

File: scripts/003-newlib.sh (hunk 1 above)
Comment (sharkwouter) [reply scripts/003-newlib.sh:5]: Fair

**Pull request merged**